### ISSUE TITLE
Disallow "browser-only" devtools in output

### ISSUE THREAD

**Issue by arichiardi** [thheller/shadow-cljs#1032]
Hi Thomas, I thought it would be good to send an idea here, but feel free to dismiss it.

The issue I posted on Slack when compiling Cloudflare workers, is:

```
[mf:err] Error: Failed to construct 'WebSocket': the constructor is not implemented.
    at Object.construct (/home/cokap/git/arudik-website/cloudflare-workers/backblaze-download-token-to-secrets/node_modules/@miniflare/web-sockets/src/plugin.ts:37:19)
    at shadow$cljs$devtools$client$websocket$start (/home/cokap/git/arudik-website/cloudflare-workers/.cljs/cljs-runtime/shadow/cljs/devtools/client/websocket.cljs:12:16)
    at shadow.cljs.devtools.client.shared.Runtime.attempt_connect_BANG_ (/home/cokap/git/arudik-website/cloudflare-workers/.cljs/cljs-runtime/shadow/cljs/devtools/client/shared.cljs:324:25)
    at Object.shadow$cljs$devtools$client$shared$init_runtime_BANG_ [as init_runtime_BANG_] (/home/cokap/git/arudik-website/cloudflare-workers/.cljs/cljs-runtime/shadow/cljs/devtools/client/shared.cljs:509:24)
    at ../.cljs/cljs-runtime/shadow.cljs.devtools.client.browser.js (/home/cokap/git/arudik-website/cloudflare-workers/.cljs/cljs-runtime/shadow/cljs/devtools/client/browser.cljs:337:3)
```

This seems to go away if I remove the import for `shadow.cljs.devtools.client.browser.js`. I know that we cannot anticipate all the crazy JS tooling out there but I was wondering if we could add something like

```clojure
:devtools {:enabled {:browser false}}
```
in order to be more selective about what we don't want to have (I still want to connect to the REPL using the underlying `node` process).

Thank you as always (just got back to Shadow for spiking this, always nice to use it!)

**Comment by thheller** [comment]
Different devtools implementations can be picked by setting a different `:runtime` in the build config. It just defaults to the browser and in this case doesn't actually mean browser but means `WebSocket`. Which I assumed would be present on all ESM runtimes but I guess that was wrong.

You may try setting `:runtime :custom :devtools {:client-ns shadow.cljs.devtools.client.node}` in the build config. Maybe the default node implementation will just. I have my doubts though given that it assumes node and commonjs for reloading purposes. Given how finicky some runtimes can be and their widely varying capabilities it is pretty hard to have a generic one. It is not just about connecting a websocket, we need to have a proper way to eval too.

Given the `:client-ns` option though you could entirely implement it on your side but replicating what the default implementations do and adjusting accordingly to the runtimes capabilities.

**Comment by arichiardi** [comment]
I think this hint gave a good attempt to solve the problem, but in a slightly different way.

After I received an exception with `{:runtime {:custom ...}}` I decided to follow the suggestion of the error and tried:

`:runtime :node`

This worked! It seems the `wrangler` is not complaining anymore with `watch` mode.

This is simply amazing because I had actually tried normal ClojureScript + esbuild as well yesterday, which is working but more cumbersome :smile:
